src/Microsoft.AspNetCore.Routing/RouteConstraintMatcher.cs

```diff
@@ -39,7 +39,7 @@ public static class RouteConstraintMatcher
                 throw new ArgumentNullException(nameof(logger));
             }
 
-            if (constraints == null)
+            if (constraints == null || constraints.Count == 0)
```

Comment:
@pranavkm has suggested a suitable replacement for a _different_ expression: `constraints != null && constraints.Count == 0`. Stick with code in this PR.